crates/database/src/repositories/response.rs

```diff
@@ -621,7 +729,9 @@ impl ResponseRepositoryTrait for PgResponseRepository {
                            conversation_id, previous_response_id, next_response_ids,
                            usage, metadata, created_at, updated_at
                     FROM responses
-                    WHERE conversation_id = $1 AND workspace_id = $2
+                    WHERE conversation_id = $1
+                      AND workspace_id = $2
+                      AND COALESCE((metadata->>'root_response')::boolean, false) = false
```

Comment:
Query uses COALESCE((metadata->>'root_response')::boolean, false) = true but index uses metadata->>'root_response' = 'true'. PostgreSQL needs exact match to use partial index. Update query to:
metadata->>'root_response' = 'true'